adapter/thing/car_charger_test.go

```diff
@@ -0,0 +1,589 @@
+						Name:    "set charging mode",
+						Command: suite.StringMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:chargepoint/ad:2", "cmd.charging_mode.set", "chargepoint", "slow"),
+						Expectations: []*suite.Expectation{
+							suite.ExpectString("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:chargepoint/ad:2", "evt.charging_mode.report", "chargepoint", "slow"),
+						},
+					},
+					{
+						Name:    "charging mode report",
+						Command: suite.NullMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:chargepoint/ad:2", "cmd.charging_mode.get_report", "chargepoint"),
+						Expectations: []*suite.Expectation{
+							suite.ExpectString("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:chargepoint/ad:2", "evt.charging_mode.report", "chargepoint", "slow"),
+						},
+					},
+				},
+			},
+			{
+				Name: "failed charging mode routing - setter error",
+				Setup: routeCarCharger(
+					mockedchargepoint.MockEnergyAwareController().
+						MockSetChargepointChargingMode("slow", errTest, true),
+					nil,
+				),
+				Nodes: []*suite.Node{
+					{
+						Name:    "set charging mode",
+						Command: suite.StringMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:chargepoint/ad:2", "cmd.charging_mode.set", "chargepoint", "slow"),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:chargepoint/ad:2", "chargepoint"),
+						},
+					},
+				},
+			},
+			{
+				Name: "failed charging mode routing - invalid mode",
+				Setup: routeCarCharger(
+					mockedchargepoint.MockEnergyAwareController(),
+					nil,
+				),
+				Nodes: []*suite.Node{
+					{
+						Name:    "set charging mode",
+						Command: suite.StringMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:chargepoint/ad:2", "cmd.charging_mode.set", "chargepoint", "dummy"),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:chargepoint/ad:2", "chargepoint"),
+						},
+					},
+				},
+			},
+			{
+				Name: "failed charging mode routing - report error",
+				Setup: routeCarCharger(
+					mockedchargepoint.MockEnergyAwareController().
+						MockSetChargepointChargingMode("slow", nil, true).
+						MockChargepointChargingModeReport("", errTest, true),
+					nil,
+				),
+				Nodes: []*suite.Node{
+					{
+						Name:    "charging mode report",
+						Command: suite.StringMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:chargepoint/ad:2", "cmd.charging_mode.set", "chargepoint", "slow"),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:chargepoint/ad:2", "chargepoint"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	s.Run(t)
+}
+
+func TestTaskCarCharger(t *testing.T) { //nolint:paralleltest
+	s := &suite.Suite{
+		Cases: []*suite.Case{
+			{
+				Name: "Thermostat tasks",
+				Setup: taskCarCharger(
+					mockedchargepoint.MockController().
+						MockChargepointCableLockReport(true, nil, true).
```

Comment:
You did not test Charging state... Which is not implemented for tasks...

Reply:
At first I rejected the idea of charging mode reporting, but then I looked at current session report :) Fixed!